xicam/plugins/__init__.py

```diff
@@ -145,6 +145,11 @@ def getPluginByName(self, name, category="Default", timeout=150):
         return plugin
 
     def getPluginsOfCategory(self, category_name):
+        while not self.loadcomplete:
+            if threads.is_main_thread():
+                QApplication.processEvents()
+            else:
+                time.sleep(0.01)
```

Comment:
I think you might want to instead use `XicamPluginManager.attach(...)` to subscribe to notifications of new plugins, rather than blocking the load thread. If this is removed does it cause problems?